File: springapp copy/main/java/com/examly/springapp/service/BookingService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.entity.Booking;

public interface BookingService {
    Booking createBooking(int userId,Long movieId,Booking booking);
    Booking getBookingById(Long bookingId);
    boolean deleteBooking(Long bookingId);
    List<Booking> getBookingbyMovie(Long movieId);
    List<Booking> getBookingByUserId(int userId);
    List<Booking> getAllBookings();

}

File: angularapp copy/app/services/booking.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { Booking } from '../models/booking.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class BookingService {
  private apiUrl="https://8080-ebbdcaaae322305198bdafdcdedbeeccfone.premiumproject.examly.io/api/booking";

  constructor(private http:HttpClient) { }

  addBooking(movieId:number,userId:number,booking:Booking):Observable<Booking>{
    return this.http.post<Booking>(this.apiUrl+"/"+userId+"/"+movieId,booking);
  }
  getUserBookings(userId:number):Observable<Booking[]>{
    return this.http.get<Booking[]>(this.apiUrl+"/user/"+userId);
  }
  getBookingbyId(bookingId:number):Observable<Booking>{
    return this.http.get<Booking>(this.apiUrl+"/"+bookingId);
  }
  updateBooking(bookingId:number,updateBooking:Booking):Observable<Booking>{
    return this.http.put<Booking>(this.apiUrl+"/"+bookingId,updateBooking);
  }
  getAllBookings():Observable<Booking[]>{
    return this.http.get<Booking[]>(this.apiUrl);
  }
}
